app/src/main/java/com/cupcake/jobsfinder/ui/jobs/JobsViewModel.kt

```diff
@@ -19,39 +18,39 @@ class JobsViewModel @Inject constructor(
 ) : BaseViewModel() {
 
     private val _jobsUIState = MutableStateFlow(JobsUiState())
-    val jobsUIState: StateFlow<JobsUiState> = _jobsUIState
-
-    private val errors: MutableList<ErrorUiState> = mutableListOf()
+    val jobsUIState = _jobsUIState.asStateFlow()
 
     init {
         viewModelScope.launch {
-            getJobRecommended()
+            getRecommendedJobs()
         }
     }
 
-    private suspend fun getJobRecommended() {
+    private fun getRecommendedJobs() {
         _jobsUIState.update { it.copy(isLoading = true) }
 
-        viewModelScope.launch {
-            getAllJobUseCase(10)
-                .catch { throwable ->
-                    errors.add(ErrorUiState(0, throwable.message.toString()))
-                    _jobsUIState.update { it.copy(error = errors) }
-                }
-                .collect { jobs ->
-                    val recommendedJobs = jobs.map { job ->
-                        JobUiState(
-                            image = "",
-                            title = "",
-                            companyName = job.company ?: "",
-                            detailsChip = listOf(job.workType.toString(), job.jobType.toString()),
-                            location = job.jobLocation ?: "",
-                            salary = job.salary ?: ""
-                        )
-                    }
-
-                    _jobsUIState.update { it.copy(job = recommendedJobs, isLoading = false) }
+        val coroutineExceptionHandler = CoroutineExceptionHandler { _, throwable ->
+            _jobsUIState.update { it.copy(isLoading = false, error = listOf(throwable.message.toString())) }
+        }
+
+        viewModelScope.launch(Dispatchers.IO + coroutineExceptionHandler) {
+            val jobs = getAllJobUseCase(RECOMMENDED_JOB_LIMIT)
+                .map { job ->
+                    JobUiState(
```

Comment:
I say this void by idea, when when make decision to check if empty or not can you check when mapper DTO -> model domain